app/dao/notifications_dao.py

```diff
@@ -85,6 +85,31 @@ def dao_create_notification(notification):
     db.session.add(notification)
 
 
+@statsd(namespace="dao")
+@transactional
+def bulk_insert_notifications(notifications):
+    """
+    Takes a list of models.Notifications and inserts or updates the DB
+    with the list accordingly
+
+    Parameters:
+    ----------
+    notifications: lof models.Notification
+
+    Return:
+    ------
+    None
+    """
+    for notification in notifications:
+        if not notification.id:
+            notification.id = create_uuid()
+        if not notification.status:
+            notification.status = NOTIFICATION_CREATED
```

Comment:
Ideal these paths should be tested.. I actually explored this option.. What do you think??